extensions/standard-processors/tests/unit/TailFileTests.cpp

```diff
@@ -78,7 +78,245 @@ TEST_CASE("TailFileWithDelimiter", "[tailfiletest2]") {
 
   // Delete the test and state file.
   remove(TMP_FILE);
-  remove(STATE_FILE);
+  remove(std::string(std::string(STATE_FILE) + "." + id).c_str());
+}
+
+TEST_CASE("TestNewContent", "[tailFileWithDelimiterState]") {
```

Comment:
There were testcases in this file already resetting plan and running TailFile multiple times, the only things they missed to verify this behaviour is the new flag you just added to clear configured processors. 

More tests don't hurt anyway, so let's leave it as is, although modifying some of those testscases to verify state storing as well might be good, too.